OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/InAppMessagePresenter.java

```diff
@@ -14,23 +14,34 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
 
 
 class InAppMessagePresenter implements AppStateWatcher.AppStateChangedListener {
 
     private static final String TAG = InAppMessagePresenter.class.getName();
-
     private final List<InAppMessage> messageQueue = new ArrayList<>();
     private final Context context;
+    private final ScheduledExecutorService interceptorExecutor = Executors.newSingleThreadScheduledExecutor();
 
     @NonNull
     private OptimoveConfig.InAppDisplayMode displayMode;
 
+    @Nullable
+    private InAppMessageInterceptor messageInterceptor = null;
+
     @Nullable
     private Activity currentActivity;
     @Nullable
     private InAppMessageView view;
 
+    private boolean interceptionInProgress = false;
```

Comment:
What's the thread manging this? (no atomic needed/can we annotate fields as @mainthread (not sure)?)

Reply:
Updated always from UI thread